gsy_framework/sim_results/electric_blue/volume_timeseries.py

```diff
@@ -60,25 +76,23 @@ class AssetVolumeTimeSeries:
     def __init__(
             self, asset_uuid: str, asset_peak_kWh: float,
             resolution: AggregationResolution):
-
-        self.asset_uuid = asset_uuid
+        super().__init__(asset_uuid, resolution)
         self.asset_peak_kWh = asset_peak_kWh
-        self.resolution = resolution
-
-        self._asset_volume_time_series_buffer: Dict[DateTime, Dict] = {}
         self._trade_profile_generator = ForwardTradeProfileGenerator(self.asset_peak_kWh)
 
-    def update_volume_time_series(
+    def update_time_series(
             self, asset_stats: ForwardDeviceStats, product_type: AvailableMarketTypes):
-        """Update asset volume time series with the current asset stats: stats from the last 15
-        minutes."""
         self._add_total_energy_bought(asset_stats, product_type)
         self._add_total_energy_sold(asset_stats, product_type)
 
     def save_time_series(self):
         """Save asset volume time series in the DB."""
         # TODO: should be implemented.
-        raise NotImplementedError()
+
+    def _fetch_asset_time_series_from_db(
+            self, year: DateTime) -> Optional[Dict[str, Dict]]:
+        """Fetch already saved asset volume time series."""
+        # TODO: should be implemented.
```

Comment:
Yes. Actually there should be a seprate PR for this in the future. It should be done after corresponding tables are implemented.

Reply:
In GSYE-398 I used GSYE-374 as base and started already. This can be done here, too, right?

Reply:
I have 2 suggestions to tackle this problem:

1- In order for this method to be able to reach DB it needs to have access to Django model classes. As it's not possible to import model classes while gsy-framework is installed in gsy-e, I suggest you to use dependency injection for the models (This way you can avoid using pony or peewee).
Something like this:
```python
    def _fetch_asset_time_series_from_db(
            self, year: DateTime, model_class) -> Optional[Dict[str, Dict]]:
```

2- As I have done some refactoring in this PR, you may want to use this branch as your base to avoid conflicts.

Thanks a lot

Reply:
IMO you should not pass the model at all, but the asset data instead as a dependency (or, even better, as a dependency of the class itself and not of the method). That way the gsy-framework will remain free from DB-specific stuff and will only contain data processing, therefore allowing it to also operate for generating CSV results.
I also agree with Hannes on branching off GSYE-374, no need to wait for it to be merged since the database tables are more or less ready in this PR.

Reply:
@spyrostz ,
When designing this method, I tried to only fetch asset time series that need to be updated, and that's why you see this method that dynamically decides on what to fetch. And, of course we always have the option of buffering everything. If you think we should stick with the buffering strategy, we can even remove this method.

Reply:
@mrsaemir correct me if I am wrong, but as far as I see the time series data are filtered by asset_uuid and year. I assume the latter part makes things more complex, since it is deduced while updating the timeseries. Even in this case though, there is a way that will allow you to keep DB packages outside of project dependencies, and the framework code clean from any DB models: closures. You can always pass a function / method (defined in gsy-web, which has access to the DB models) with one argument (the year) as an argument in the constructor of this class, which will include the DB connection in its scope. Keep the function in a separate variable, and invoke it from `_fetch_asset_time_series_from_db` . The interface of the class will become more complex, but at least the repo will remain without DB dependencies.

Reply:
Sorry, I do not see a PR or story for this, can you please point me to this? Thanks a lot!

Reply:
Actually this will be related to the next phase of the task. After someone created gsy-web tables, he should also adapt this method with the suggestions mentioned by Spyros.

Reply:
If you think we should at least fix the signature of the function, we can do that, but I have no idea about it for now. Please let me know if you have any suggestions.

Reply:
> Actually this will be related to the next phase of the task.

So, this will include a PR to gsy-web (probably with target GSYE-374) where a new model is added, right? Are you taking care of it in the frame of the ticket? If so, then I will leave my LGTM here.

Reply:
Exactly. I will help to correctly integrate the new model to this method. Thanks